src/number/floatFromString.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * Floating point branded newtype from strings. Parameters: min, max are inclusive.
+ *
+ * Represents string floating point numbers:
+ *
+ * ```math
+ *  { f | f ∈ ℝ, f >= -Number.MAX_VALUE, f <= Number.MAX_VALUE }
+ * ```
+ *
+ * @since 1.0.0
+ */
+import { Kind, Kind2, URIS, URIS2, HKT2 } from 'fp-ts/HKT'
+import * as Enc from 'io-ts/Encoder'
+import * as D from 'io-ts/Decoder'
+import * as Eq_ from 'fp-ts/Eq'
+import * as G from 'io-ts/Guard'
+import * as TD from 'io-ts/TaskDecoder'
+import * as t from 'io-ts/Type'
+import * as N from 'fp-ts/number'
+import { flow } from 'fp-ts/function'
+import { Type as Type_ } from 'io-ts'
+
+import * as Arb from '../internal/ArbitraryBase'
+import * as float from './float'
+
+/**
+ * @since 1.0.0
+ * @category Model
+ */
+export type SchemableParams<S> = (
+  params?: float.FloatParams
+) => HKT2<S, string, float.Float>
+
+/**
+ * @since 1.0.0
+ * @category Model
+ */
+export type SchemableParams1<S extends URIS> = (
+  params?: float.FloatParams
+) => Kind<S, float.Float>
+
+/**
+ * @since 1.0.0
+ * @category Model
+ */
+export type SchemableParams2<S extends URIS2> = (
+  params?: float.FloatParams
+) => Kind2<S, string, float.Float>
+
+/**
+ * @since 1.0.0
+ * @category Model
+ */
+export type SchemableParams2C<S extends URIS2> = (
+  params?: float.FloatParams
+) => Kind2<S, unknown, float.Float>
+
+/**
+ * @since 1.0.0
+ * @category Instances
+ */
+export const Decoder: SchemableParams2C<D.URI> = params => ({
+  decode: flow(Number, float.Decoder(params).decode),
```

Comment:
I had that idea originally, but I think the number constructor has stronger condition.  Here's a comment @newswim made about it:

https://github.com/jacob-alford/schemata-ts/pull/41#discussion_r974908510

Reply:
And you're right about checking for a string first, I need to think of a good way to handle that, but io-ts/Guard > number does a NaN check which I'm using

Reply:
imo if I got "1.2.3.4" as input I would be perplexed if I got "1.2" as the float output, instead of an error

Reply:
wait, I think Dan's comment is backwards:

![image](https://user-images.githubusercontent.com/387382/198064110-f31e4198-6d0c-4f05-91af-da3763822dc3.png)

Reply:
I don't think `parseFloat` takes a base argument; that's only relevant for `intFromString`

Reply:
I interpreted Dan's comment as Number turns more things into NaN which will be guarded against in the implementation, which is a good thing IMO.  And I think we might be in agreement there

Reply:
https://camchenry.com/blog/parsefloat-vs-number

I think the `Number` constructor is marginally better than `parseFloat`, mostly because it also parses number literals like binary, hexadecimal, and octal.  Though it's a bit annoying that it parses empty string (and others) as 0, and true as 1

Reply:
yeah, sorry, I agree `Number` is better. I really don't know how I managed to completely misread Dan's comment. several times.

> Though it's a bit annoying that it parses empty string (and others) as 0, and true as 1

we should handle those cases here and decode to error, imo

Reply:
Agreed!